receiver/redisreceiver/pdata.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package redisreceiver
+
+import (
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+func newResourceMetrics(ms pdata.MetricSlice, serviceName string) pdata.ResourceMetrics {
+	rm := pdata.NewResourceMetrics()
+	r := rm.Resource()
+	rattrs := r.Attributes()
+	rattrs.Insert("type", pdata.NewAttributeValueString(typeStr))
+	rattrs.Insert("service.name", pdata.NewAttributeValueString(serviceName))
+	ilm := pdata.NewInstrumentationLibraryMetrics()
+	rm.InstrumentationLibraryMetrics().Append(ilm)
+	ms.CopyTo(ilm.Metrics())
+	return rm
+}
+
+func buildKeyspaceTriplet(k *keyspace, t *timeBundle) pdata.MetricSlice {
+	keysPDM := buildKeyspaceKeysMetric(k, t)
+	expiresPDM := buildKeyspaceExpiresMetric(k, t)
+	ttlPDM := buildKeyspaceTTLMetric(k, t)
+
+	ms := pdata.NewMetricSlice()
+	ms.Append(keysPDM)
+	ms.Append(expiresPDM)
+	ms.Append(ttlPDM)
+
+	return ms
+}
+
+func buildKeyspaceKeysMetric(k *keyspace, t *timeBundle) pdata.Metric {
+	m := &redisMetric{
+		name:   "redis/db/keys",
+		labels: map[string]string{"db": k.db},
+		pdType: pdata.MetricDataTypeIntGauge,
+	}
```

Comment:
This can be a constant to avoid unnecessary allocation.

Reply:
Nice, thanks. This is going to be replaced by your suggestion below (https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/1889#discussion_r551596159) in a followup PR, which should take care of the unnecessary allocation.